docusaurus/docs/dev-docs/migration/v4-to-v5/breaking-changes/redux-content-manager-app-state.md

```diff
@@ -0,0 +1,56 @@
+---
+title: The ContentManagerAppState redux is modified
+description: In Strapi 5, the redux store for the Content Manager has been changed and some redux actions were removed.
+sidebar_label: ContentManagerAppState redux modified
+displayed_sidebar: devDocsMigrationV5Sidebar
+tags:
+ - breaking changes
+ - Content Manager
+ - redux
+---
+
+import Intro from '/docs/snippets/breaking-change-page-intro.md'
+import MigrationIntro from '/docs/snippets/breaking-change-page-migration-intro.md'
+
+# The ContentManagerAppState redux is modified
+
+In Strapi 5, the redux store for the Content Manager has been changed and some redux actions were removed. Notably, the `useContentManagerInitData` redux state for the Content Manager has been refactored to remove `ModelsContext`. Users might be relying on the original structure in a middleware or subscriber; doing so this will break their application.
+
+<Intro />
+
+## Breaking change description
+
+<SideBySideContainer>
+
+<SideBySideColumn>
+
+**In Strapi v4**
+
+The redux store can fire the following actions:
+
+- `'ContentManager/App/RESET_INIT_DATA’`
+- `'ContentManager/App/GET_INIT_DATA’`
+- `'ContentManager/App/SET_INIT_DATA’`
+
+</SideBySideColumn>
+
+<SideBySideColumn>
+
+**In Strapi 5**
+
+The redux store no longer fires the following actions:
+
+- `'ContentManager/App/RESET_INIT_DATA’`
+- `'ContentManager/App/GET_INIT_DATA’`
+
+</SideBySideColumn>
+
+</SideBySideContainer>
+
+## Migration
+
+<MigrationIntro />
+
+### Manual procedure
+
+The redux store in Strapi 5 continues to fire `'ContentManager/App/SET_INIT_DATA’`, so users should instead listen for this action in their middlewares only.
```

Comment:
I think we should also mention that the structure of the data in the payload has changed.

V4
```
interface SetInitDataAction {
  type: typeof SET_INIT_DATA;
  data: {
    authorizedCollectionTypeLinks: ContentManagerAppState['collectionTypeLinks'];
    authorizedSingleTypeLinks: ContentManagerAppState['singleTypeLinks'];
    components: ContentManagerAppState['components'];
    contentTypeSchemas: ContentManagerAppState['models'];
    fieldSizes: ContentManagerAppState['fieldSizes'];
  };
}
```

V5
```
interface SetInitDataAction {
  type: typeof SET_INIT_DATA;
  authorizedCollectionTypeLinks: ContentManagerAppState['collectionTypeLinks'];
  authorizedSingleTypeLinks: ContentManagerAppState['singleTypeLinks'];
  components: ContentManagerAppState['components'];
  contentTypeSchemas: ContentManagerAppState['models'];
  fieldSizes: ContentManagerAppState['fieldSizes'];

}
```

e.g. we are no longer nesting the attributes in 'data'

Reply:
Thank you, Jamie!
I [added it.](https://github.com/strapi/documentation/pull/1995/commits/c7be31c24806592ad4fc77284bb458f4d1460812)